infra/src/dsl/value-files-generators/local-setup.ts

```diff
@@ -42,24 +43,58 @@ const mapServiceToNXname = async (serviceName: string) => {
     )
   return nxName.length === 1 ? nxName[0] : serviceName
 }
-
 export const getLocalrunValueFile = async (
   runtime: Localhost,
   services: Services<LocalrunService>,
 ): Promise<LocalrunValueFile> => {
-  const dockerComposeServices: Services<LocalrunService> = await Object.entries(
-    services,
-  ).reduce(async (acc, [name, service]) => {
-    return {
-      ...(await acc),
-      [name]: ` ${
-        runtime.ports[name] ? `PORT=${runtime.ports[name]} ` : ''
-      } PROD_MODE=true ${Object.entries(service.env)
-        .filter(([name, val]) => !EXCLUDED_ENVIRONMENT_NAMES.includes(name))
-        .map(([key, value]) => `${key}="${value}"`)
-        .join(' ')} yarn start ${await mapServiceToNXname(name)}`,
-    }
-  }, Promise.resolve({}))
+  const dockerComposeServices = await Object.entries(services).reduce(
+    async (acc, [name, service]) => {
+      const portConfig = runtime.ports[name]
+        ? { PORT: runtime.ports[name].toString() }
+        : {}
+      const serviceNXName = await mapServiceToNXname(name)
+      return {
+        ...(await acc),
+        [name]: {
+          env: Object.assign(
+            {},
+            Object.entries(service.env)
+              .filter(
+                ([name, val]) => !EXCLUDED_ENVIRONMENT_NAMES.includes(name),
+              )
+              .reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {}),
+            { PROD_MODE: 'true' },
+            portConfig,
+          ) as Record<string, string>,
+          command: `(source ${join(
+            rootDir,
+            '.env.' + serviceNXName,
+          )} && yarn start ${serviceNXName})`,
+        },
+      }
+    },
+    Promise.resolve(
+      {} as {
+        [name: string]: { env: Record<string, string>; command: string }
+      },
+    ),
+  )
+
+  // dump all env values to files
+  await Promise.all(
+    Object.entries(dockerComposeServices).map(async ([name, svc]) => {
+      const serviceNXName = await mapServiceToNXname(name)
+      await writeFile(
+        join(rootDir, `.env.${serviceNXName}`),
+        Object.entries(svc.env)
+          .map(
+            ([name, value]) => `export ${name}="${value.replace(/"/g, '\\"')}"`,
```

Comment:
I think we need more handling here. Some values might contain backslashes, and even be multi-line json. Using `'` instead of `"` would be better. We don't know what escape characters might lure in some secrets.

Reply:
Aaa, of course. Well spotted. Fixing it.